derive_client/data_types/models.py

```diff
@@ -1,13 +1,50 @@
 """Models used in the bridge module."""
 
-from pydantic import BaseModel, ConfigDict
 
+from dataclasses import dataclass
+
+from derive_action_signing.module_data import ModuleData
+from derive_action_signing.utils import decimal_to_big_int
+from eth_abi.abi import encode
+from pydantic import BaseModel, ConfigDict
+from web3 import Web3
 
 from .enums import ChainID, Currency
 
 Address = str
 
 
+@dataclass
+class CreateSubAccountDetails:
+    amount: int
+    base_asset_address: str
+    sub_asset_address: str
+
+    def to_eth_tx_params(self):
+        return (
+            decimal_to_big_int(self.amount),
+            Web3.to_checksum_address(self.base_asset_address),
+            Web3.to_checksum_address(self.sub_asset_address),
+        )
+
+
+@dataclass
+class CreateSubAccountData(ModuleData):
+    amount: int
+    asset_name: str
+    margin_type: str
+    create_account_details: CreateSubAccountDetails
+
+    def to_abi_encoded(self):
+        return encode(
+            ['uint256', 'address', 'address'],
+            self.create_account_details.to_eth_tx_params(),
+        )
+
+    def to_json(self):
+        return {}
```

Comment:
these were duplicated:
- in [`create_subaccount_module.py `](https://github.com/8ball030/derive_client/blob/main/derive_client/create_subaccount_module.py) (with incorrect import), and
- in [`base_client`](https://github.com/8ball030/derive_client/blob/10085060398298cbf5475ee0c5b7b210585beb61/derive_client/base_client.py#L55-L83). I moved it to the `data_types/models.py` module

Reply:
this should be importing from the signing module.

Reply:
Looks like dead code was found